graph/src/components/adapter.rs

```diff
@@ -0,0 +1,762 @@
+use std::{
+    collections::HashMap,
+    ops::{Add, Deref},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use chrono::{DateTime, Duration, Utc};
+
+use itertools::Itertools;
+use slog::{o, warn, Discard, Logger};
+use thiserror::Error;
+
+use crate::{
+    blockchain::{BlockHash, ChainIdentifier},
+    cheap_clone::CheapClone,
+    data::value::Word,
+    prelude::error,
+    tokio::sync::RwLock,
+};
+
+use crate::components::store::{BlockStore as BlockStoreTrait, ChainStore as ChainStoreTrait};
+
+const VALIDATION_ATTEMPT_TTL_SECONDS: i64 = 60 * 5;
```

Comment:
Would be better to make that a `Duration`. That'll also save you the `_SECONDS` postfix ;)

Reply:
![image](https://github.com/graphprotocol/graph-node/assets/1640325/5f154cdc-19aa-490c-80a7-90c11909897f)